server/src/services/SearchService.js

```diff
@@ -71,7 +72,13 @@ async function search(info) {
     query = filterQuery;
   }
 
-  return PostModel.find(query);
+  const posts = await PostModel.find(query);
+
+  const types = posts.map((p) => p.type);
+
+  const forms = await FormModel.find({ type: { $in: types } }, { type: 1, content: 1 });
```

Comment:
This can be inefficient for large collection sizes. We can create index for the type field of forms collection.